### PR TITLE
Feature/stable repo links

### PR DESCRIPTION
**Pull request by chrpr**
@barnabyalter @NYULibraries/finding-aids 

Hi All,

This replaces closed PR #48. I've backed out changes to coffeescript and routes for dropdown selection that were deemed unnecessary following our last sprint planning meeting. I also re-wrote a few of the tests and rebased on the current development branch.

Let me know if there's anything else you'd like to see addressed here.

-Corey

### CODE REVIEW

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -66,9 +66,9 @@ class CatalogController < ApplicationController
...
     # solr fields to be displayed in the index (search results) view
     #   The ordering of the field names is the order of the display
-    config.add_index_field solr_name("repository",        :stored_sortable), label: "Library", helper_method: :render_repository_facet_link
-    config.add_index_field solr_name("unittitle",         :displayable),  :label => "Title", :helper_method => :render_field_item
-    config.add_index_field solr_name("abstract",          :displayable),  :label => "Abstract", :helper_method => :render_field_item
+    config.add_index_field solr_name("repository",        :stored_sortable), label: "Library", helper_method: :render_repository_link
+    config.add_index_field solr_name("unittitle",         :displayable),  :label => "Title", :highlight => true, :helper_method => :render_field_item
+    config.add_index_field solr_name("abstract",          :displayable),  :label => "Abstract", :highlight => true, :helper_method => :render_field_item
     config.add_index_field solr_name("format",            :displayable),  :label => "Format", :helper_method => :render_field_item
```
Comment (barnabyalter) [inline app/controllers/catalog_controller.rb]: @chrpr These index fields no longer have highlighting , please removed them

File: app/helpers/results_helper.rb [hunk 2]
```diff
@@ -20,9 +19,13 @@ def render_series_facet_link(doc)
...
 
   def render_repository_facet_link(doc)
+    repository_label repositories.find{|key,hash| hash["admin_code"] == doc}[1]["url"]
+  end
+  
+  def render_repository_link(doc)
     repos_id = Solrizer.solr_name("repository", :stored_sortable)
     if doc.is_a?(Hash) && doc[:document].present? && doc[:document][repos_id].present?
-      link_to_repository repositories[doc[:document][repos_id]]["admin_code"]
+      link_to_repository repositories.find{|key,hash| hash["admin_code"] == doc[:document][repos_id]}[1]["url"]
```
Comment (barnabyalter) [inline app/helpers/results_helper.rb]: @chrpr Could you add a comment describing what you're doing here

File: config/routes.rb [hunk 3]
```diff
@@ -6,7 +6,9 @@
 
   # Create named routes for each collection specified in tabs.yml
   YAML.load_file( File.join(Rails.root, "config", "repositories.yml") )["Catalog"]["repositories"].each do |coll|
```
Comment (barnabyalter) [inline config/routes.rb]: @chrpr I still think you should use the Repositories class here:

https://github.com/NYULibraries/findingaids/blob/development/app/controllers/catalog_controller.rb#L114

**Comment by chrpr** [comment]
@barnabyalter Just pushed 3 new commits, that I think address your remaining comments. Let me know if there's anything else you'd like out of me on this.

**Pull request merged**